File: guanabara-java/src/com/danillo/algoritmos/lista08/Exercicio82.java
package com.danillo.algoritmos.lista08;

import java.util.Scanner;

public class Exercicio82 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);
        float[] vetor = new float[10];

        float somaNotas = 0, maiorNota=0;
        for(int i = 0; i < vetor.length; i++){
            System.out.println("Insira uma nota: ");
            vetor[i] = entrada.nextFloat();

            somaNotas = somaNotas + vetor[i];

            if(vetor[i] > maiorNota){
                maiorNota = vetor[i];
            }
        }
        float mediaNotas = somaNotas/vetor.length;

        int alunoAcimaMedia = 0;
        for(int i = 0; i< vetor.length; i++){
            if(vetor[i] > mediaNotas){
                alunoAcimaMedia++;
            }
        }

        System.out.println("Média das notas da turma: " +mediaNotas+
                           "\nAlunos acima da média da turma: " +alunoAcimaMedia+
                           "\nMaior nota da turma: " +maiorNota);

        System.out.print("Posição do vetor com a maior nota: ");
        for(int i = 0; i < vetor.length; i++){
            if(vetor[i] == maiorNota){
                System.out.print(i+ " ");
            }
        }
        System.out.println();
    }
}

File: guanabara-python/src/lista08/exercicio82.py
vetor = []
soma_nota = 0
quantidade_notas = 0
maior_nota = 0
alunos_acima_media = 0

for i in range(5):
    nota = float(input("Insira a nota: "))
    vetor.append(nota)

    if nota > maior_nota:
        maior_nota = nota

    soma_nota = soma_nota + nota
    quantidade_notas = quantidade_notas + 1

media_nota = soma_nota / quantidade_notas

for i in range(5):
    if vetor[i] > media_nota:
        alunos_acima_media = alunos_acima_media + 1

print(f"Média: {media_nota}.\nQuantidade de alunos acima da média: {alunos_acima_media}.\n"
      f"Maior nota: {maior_nota}.")

print("Posição da maior nota:")
for i in range(5):
    if vetor[i] == maior_nota:
        print(f"Nota: {vetor[i]} | Posição: {i}")
